backend/database.py

```diff
@@ -0,0 +1,59 @@
+#
+from sqlite3 import PARSE_DECLTYPES, connect
+
+db = connect('casino.db', detect_types=PARSE_DECLTYPES, check_same_thread=False)
+db.row_factory = lambda c, r: {k: v for k, v in zip([cl[0] for cl in c.description], r)} 
+c = db.cursor()
+
+
+def get_id(uid):
+    game = c.execute(
+              'SELECT * FROM game where name=(?) order by modified'
+              , [uid]
+              ).fetchone()
+    return game
+
+
+def update(uid, blob):
+    INSERT_GAME = f'INSERT OR REPLACE into game(name) values("{uid}");'
```

Comment:
In order to avoid an sql injection you should let `execute` [bind the parameters](https://docs.python.org/3/library/sqlite3.html#sqlite3.Cursor.execute). 
```
c.execute("INSERT OR REPLACE into game(name) values(?)", (name,))
```

Reply:
>  I'm wondering if the real conversion should all be handled within State class, or if this is something that would live more as helper functions in Card & Unit, or there is something to leverage in starlette. Let me know what approach you think would make most sense here.

Yeah, I agree, I was thinking it would be convenient if `State` knew how to serialize/deserialize.
```
class State:
    ...
    def to_json(self):
        ...
    @classmethod    
    def from_json(cls, raw):
       ...
```

And then to save or load state in `api.py` where we currently do `GAMES[id] = state` we could do something along the lines of `save_game_state(state.game_id, state.to_json())`.